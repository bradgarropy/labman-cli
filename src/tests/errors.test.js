const {
    errorTokenNotFound,
    errorInvalidToken,
    errorLoginFailed,
    errorRepoNotFound,
    errorLabelExists,
} = require("../errors")

let logs
const consoleLog = console.log

beforeEach(() => {
    logs = []
    console.log = message => logs.push(message)
})

afterEach(() => (console.log = consoleLog))

describe("token", () => {
    test("not found", () => {
        errorTokenNotFound()
        expect(logs).toMatchSnapshot()
    })

    test("invalid", () => {
        errorInvalidToken()
        expect(logs).toMatchSnapshot()
    })
})

describe("login ", () => {
    test("failed", () => {
        errorLoginFailed()
        expect(logs).toMatchSnapshot()
    })
})

describe("repo not found", () => {
    test("with string", () => {
        errorRepoNotFound("bradgarropy/labman-cli")
        expect(logs).toMatchSnapshot()
    })

    test("without string", () => {
        errorRepoNotFound("")
        expect(logs).toMatchSnapshot()
    })
})

describe("label exists", () => {
    test("with string", () => {
        errorLabelExists("todo")
        expect(logs).toMatchSnapshot()
    })

    test("without string", () => {
        errorLabelExists("")
        expect(logs).toMatchSnapshot()
    })
})
