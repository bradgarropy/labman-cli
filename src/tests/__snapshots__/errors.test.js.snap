// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`label exists with string 1`] = `
Array [
  "[1m[91m x Label[1m[96m todo [39m[91m[22m[1malready exists![39m[22m",
]
`;

exports[`label exists without string 1`] = `
Array [
  "[1m[91m x Label already exists![39m[22m",
]
`;

exports[`login  failed 1`] = `
Array [
  "",
  "[91mLogin failed![39m Please try again.",
]
`;

exports[`repo not found with string 1`] = `
Array [
  "",
  "[1m[91mRepository[1m[96m bradgarropy/labman-cli [39m[91m[22m[1mdoes not exist![39m[22m",
]
`;

exports[`repo not found without string 1`] = `
Array [
  "",
  "[1m[91mRepository does not exist![39m[22m",
]
`;

exports[`token invalid 1`] = `
Array [
  "",
  "[91mInvalid token![39m Please run the [96mlogin[39m command again.",
  "",
  "[96mlabman login <username> <token>[39m",
]
`;

exports[`token not found 1`] = `
Array [
  "",
  "You are not logged in, please run the [96mlogin[39m command.",
  "",
  "[96mlabman login <username> <token>[39m",
]
`;
